docs/high-availability.md

```diff
@@ -0,0 +1,196 @@
+# High Availability Kubernetes Clusters
+
+## Introduction
+This document describes how to build a high-available Kubernetes cluster.  This is a fairly advanced topic.
+Users who merely want to experiment with Kubernetes are encouraged to use configurations that are simpler to set up such as
+the simple [Docker based single node cluster instructions](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/docker.md),
+or try [Google Container Engine](https://cloud.google.com/container-engine/) for hosted Kubernetes.
+
+Also, at this time high availability support for Kubernetes is not continuously tested in our end-to-end (e2e) testing.  We will
+be working to add this continuous testing, but for now the single-node master installations are more heavily tested.
+
+## Overview
+Setting up a truly reliable, highly available distributed system requires a number of steps, it is akin to
+wearing underwear, pants, a belt, suspenders, another pair of underwear, and another pair of pants.  We go into each
```

Comment:
this is amusing. i'd say it's more like keeping a spare outfit in your backpack and not having anywhere discrete to change!